httpbakery/agent/agent.go

```diff
@@ -123,11 +124,21 @@ func LoginCookie(req *http.Request) (username string, key *bakery.PublicKey, err
 	return al.Username, al.PublicKey, nil
 }
 
-// agent represents an agent that can be used for agent authentication.
+// agent is the internal version of the agent type which also
+// includes the parsed URL.
 type agent struct {
-	pathParts []string
-	username  string
-	key       *bakery.KeyPair
+	url *url.URL
+	Agent
+}
+
+// agent represents an agent that can be used for agent authentication.
+type Agent struct {
+	// URL holds the URL associated with the agent.
+	URL string
+	// Username holds the username to use for the agent.
+	Username string
+	// Key holds the agent's private key pair.
+	Key *bakery.KeyPair
```

Comment:
Using a KeyPair representation will be awkward if you want to use the same type for serializing as an agent-cookie.